Objective-C/CBLReplicatorConfiguration.m

```diff
@@ -139,16 +148,29 @@ - (void) setNetworkInterface: (NSString*)networkInterface {
 - (void) setDocumentIDs: (NSArray<NSString *>*)documentIDs {
     [self checkReadonly];
     _documentIDs = documentIDs;
+    _defaultCollectionConfig.documentIDs = documentIDs;
```

Comment:
The documentID needs to link with the one in the _collectionConfigs. And if there is no default config in the dictionary, an exception will be thrown.

```
- (void) setDocumentIDs: (NSArray<NSString *>*)documentIDs {
    auto config = _collectionConfigs[_defaultCollection]; // Need to make sure that the collection can be key (e.g. implement hash and etc).
    if (!config) {
        // Throw exception
    }
    config.documentIDs = documentIDs
}

- (NSArray<NSString *>*) getDocumentIDs {
    auto config = _collectionConfigs[_defaultCollection]; // Need to make sure that the collection can be key (e.g. 
    return config.documentIDs;
}
```

Reply:
Then, there is no need to have _documentIDs as well.